### PR TITLE
UBERF-7495: Global editor kit extensions

### LINKED ISSUE #5988: Support action items in all editors
<sub><a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2Njg1MjY2Yjc0ZmRjOGExYzc1OGI3NDciLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.a_sAa8SWjYn8Zfjk6npK69kmb3jvflQeBuLmFhnizNU">Huly&reg;: <b>UBERF-7495</b></a></sub>

### PR DESCRIPTION
**Pull request by lexiv0re**
* Introduces global editor kit extensions
* Moves Action items extensions to the new approach

Note: it was decided to keep the server editor extensions hardcoded as it's gonna be reworked soon anyways.

Related to: https://front.hc.engineering/workbench/platform/tracker/UBERF-7495

Closes https://github.com/hcengineering/platform/issues/5988

<sub><a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NjkwMjAxNDE2OGQ5ZTUzMzk0MTU5ZjIiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.fBfOl2zlqCevIFX2ZCiX96VN5XiVDuJuBOBci3qrNIc">Huly&reg;: <b>UBERF-7577</b></a></sub>

### CODE REVIEW

File: plugins/controlled-documents-resources/package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
     "ts-jest": "^29.1.1",
     "@types/jest": "^29.5.5",
     "eslint-plugin-svelte": "^2.35.1",
-    "svelte-eslint-parser": "^0.33.1"
+    "svelte-eslint-parser": "^0.33.1",
+    "yjs": "^13.5.52"
```
Comment (haiodo) [inline plugins/controlled-documents-resources/package.json]: why we need it?

File: plugins/controlled-documents-resources/src/components/document/DocumentDiffViewer.svelte [hunk 2]
```diff
@@ -1,15 +1,15 @@
 <script lang="ts">
   import { getContext, onDestroy } from 'svelte'
-  import { Doc as Ydoc } from 'yjs'
+  import type { Doc as Ydoc } from 'yjs'
```
Comment (haiodo) [inline plugins/controlled-documents-resources/src/components/document/DocumentDiffViewer.svelte]: Do not see usage

File: plugins/text-editor-assets/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
+// Copyright © 2021 Hardcore Engineering Inc.
```
Comment (haiodo) [inline plugins/text-editor-assets/src/index.ts]: 2024?

File: plugins/controlled-documents-resources/package.json (hunk 1 above)
Comment (lexiv0re) [reply plugins/controlled-documents-resources/package.json]: We import type for YDoc for documents diff view. It was in dependencies and I moved it to devDependencies as only used for types.

File: plugins/controlled-documents-resources/src/components/document/DocumentDiffViewer.svelte (hunk 2 above)
Comment (lexiv0re) [reply plugins/controlled-documents-resources/src/components/document/DocumentDiffViewer.svelte]: It's there
image

File: plugins/controlled-documents-resources/package.json (hunk 1 above)
Comment (haiodo) [reply plugins/controlled-documents-resources/package.json]: Do we really need it?

File: plugins/text-editor-assets/src/index.ts (hunk 3 above)
Comment (lexiv0re) [reply plugins/text-editor-assets/src/index.ts]: Updated

File: plugins/controlled-documents-resources/package.json (hunk 1 above)
Comment (lexiv0re) [reply plugins/controlled-documents-resources/package.json]: Hid it in text-editor and removed yjd dependency from controlled documents.

**Review by haiodo — APPROVED** [review]

**Pull request merged**